packages/@eventual/core-runtime/src/local/stores/bucket-store.ts

```diff
@@ -286,14 +288,51 @@ export class LocalBucketStore implements BucketStore, LocalSerializable {
     };
   }
 
-  public generatePresignedUrl(
-    _bucketName: string,
-    _key: string,
-    _operation: PresignedUrlOperation,
-    _expires?: DurationSchedule | undefined
+  public async generatePresignedUrl(
+    bucketName: string,
+    key: string,
+    operation: PresignedUrlOperation,
+    expires?: DurationSchedule | undefined
   ): Promise<BucketGeneratePresignedResult> {
     // https://github.com/functionless/eventual/issues/341
-    throw new Error("Presigned urls are not supported in Eventual Local");
+    const expireDuration = expires
+      ? computeDurationSeconds(expires)
+      : // default 1 hour
+        3_600;
+    if (expireDuration > 604800) {
+      throw new Error(
+        "Presigned URL expiration must be less than or equal to 7 days (604800 seconds)"
+      );
+    }
+    const expiration = new Date(
+      this.props.localConnector.getTime().getTime() + expireDuration * 1000
+    );
+    const data: PresignUrlEnvelope = {
+      bucketName,
+      key,
+      operation,
+      expires: expiration.toISOString(),
+    };
+
+    return {
+      url: `${this.props.baseUrl}/__bucket/presigned/${Buffer.from(
+        JSON.stringify(data)
+      ).toString("base64url")}`,
+      expires: expiration.toISOString(),
+    };
+  }
+
+  public decodeAsyncUrlKey(
```

Comment:
I don't understand the name "decode async url key"

Reply:
lol, I let the AI name this and forgot to change it.